src/main/drivers/serial_uart_stm32h7xx.c

```diff
@@ -0,0 +1,567 @@
+        .txBufferSize = ARRAYLEN(uart6TxBuffer),
+        .rxBufferSize = ARRAYLEN(uart6RxBuffer),
+    },
+#endif
+
+#ifdef USE_UART7
+    {
+        .device = UARTDEV_7,
+        .reg = UART7,
+#ifdef USE_DMA
+        .rxDMARequest = DMA_REQUEST_UART7_RX,
+        .rxDMAStream = UART7_RX_DMA_STREAM,
+        .txDMARequest = DMA_REQUEST_UART7_TX,
+        .txDMAStream = UART7_TX_DMA_STREAM,
+#endif
+        .rxPins = {
+            { DEFIO_TAG_E(PA8), GPIO_AF11_UART7 },
+            { DEFIO_TAG_E(PB3), GPIO_AF11_UART7 },
+            { DEFIO_TAG_E(PE7), GPIO_AF7_UART7 },
+            { DEFIO_TAG_E(PF6), GPIO_AF7_UART7 },
+        },
+        .txPins = {
+            { DEFIO_TAG_E(PA15), GPIO_AF11_UART7 },
+            { DEFIO_TAG_E(PB4),  GPIO_AF11_UART7 },
+            { DEFIO_TAG_E(PE8),  GPIO_AF7_UART7 },
+            { DEFIO_TAG_E(PF7),  GPIO_AF7_UART7 },
+        },
+        .rcc_apb1 = RCC_APB1L(UART7),
+        .rxIrq = UART7_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART7_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART7,
+        .txBuffer = uart7TxBuffer,
+        .rxBuffer = uart7RxBuffer,
+        .txBufferSize = ARRAYLEN(uart7TxBuffer),
+        .rxBufferSize = ARRAYLEN(uart7RxBuffer),
+    },
+#endif
+
+#ifdef USE_UART8
+    {
+        .device = UARTDEV_8,
+        .reg = UART8,
+#ifdef USE_DMA
+        .rxDMARequest = DMA_REQUEST_UART8_RX,
+        .rxDMAStream = UART8_RX_DMA_STREAM,
+        .txDMARequest = DMA_REQUEST_UART8_TX,
+        .txDMAStream = UART8_TX_DMA_STREAM,
+#endif
+        .rxPins = {
+            { DEFIO_TAG_E(PE0), GPIO_AF8_UART8 }
+        },
+        .txPins = {
+            { DEFIO_TAG_E(PE1), GPIO_AF8_UART8 }
+        },
+        .rcc_apb1 = RCC_APB1L(UART8),
+        .rxIrq = UART8_IRQn,
+        .txPriority = NVIC_PRIO_SERIALUART8_TXDMA,
+        .rxPriority = NVIC_PRIO_SERIALUART8,
+        .txBuffer = uart8TxBuffer,
+        .rxBuffer = uart8RxBuffer,
+        .txBufferSize = ARRAYLEN(uart8TxBuffer),
+        .rxBufferSize = ARRAYLEN(uart8RxBuffer),
+    },
+#endif
+};
+
+#ifdef USE_DMA
+static void handleUsartTxDma(uartPort_t *s);
+#endif
+
+void uartIrqHandler(uartPort_t *s)
+{
+    UART_HandleTypeDef *huart = &s->Handle;
+    /* UART in mode Receiver ---------------------------------------------------*/
+    if ((__HAL_UART_GET_IT(huart, UART_IT_RXNE) != RESET)) {
+        uint8_t rbyte = (uint8_t)(huart->Instance->RDR & (uint8_t) 0xff);
+
+        if (s->port.rxCallback) {
+            s->port.rxCallback(rbyte, s->port.rxCallbackData);
+        } else {
+            s->port.rxBuffer[s->port.rxBufferHead] = rbyte;
+            s->port.rxBufferHead = (s->port.rxBufferHead + 1) % s->port.rxBufferSize;
+        }
+        CLEAR_BIT(huart->Instance->CR1, (USART_CR1_PEIE));
+
+        /* Disable the UART Error Interrupt: (Frame error, noise error, overrun error) */
+        CLEAR_BIT(huart->Instance->CR3, USART_CR3_EIE);
+
+        __HAL_UART_SEND_REQ(huart, UART_RXDATA_FLUSH_REQUEST);
```

Comment:
Is this necessary? It seems flush is needed only when stopping reception

Reply:
Added an item to improvement list.